### ISSUE TITLE
A little question about the code

### ISSUE THREAD

**Issue by emrekeles-arch** [davidmoraleees/Breast-image-preprocessing-for-mammographic-tissue-segmentation#1]
Thank you very much for your hard work. I have a question: Could it be normal for my code to get stuck at the np.isclose() line in the find_intersection function?

**Comment by davidmoraleees** [comment]
Hi,
Are you encountering any errors in the terminal? If so, could you share them?
Do you have the numpy package version 2.1.0 or above installed?

**Comment by emrekeles-arch** [comment]
I don't actually get an error, it's just that part of the code takes too long ?

**Comment by davidmoraleees** [comment]
Yes, I regret to inform you that it takes plenty of time to process an image. I suggest you leave the code running while you are doing other things.  Maybe you can speed it up by running the code on your GPU, but I haven't tried that on this particular project.
How much time is it taking? When I processed all the images from the dataset, it took several hours to be done.

**Comment by emrekeles-arch** [comment]
I worked on the CPU in Colab and couldn't process a single image for about 1 hour. The code kept spinning on the np.isclose() line

**Comment by davidmoraleees** [comment]
I have set up a new environment on a new PC and have installed the requirements.txt. I have executed the main.py script to process one pair of CC and MLO images and haven't got any issues. It took about 5 minutes to be completed. I am working on VSCode.

**Comment by emrekeles-arch** [comment]
Then the problem may be related to my data set or environment. Thank you so much for replying.

**Comment by davidmoraleees** [comment]
Your welcome. If you have more problems let me know.

**Comment by emrekeles-arch** [comment]
As far as I can tell, you are only using photos taken from the right, but I also have LCC and LMLO images. Does this require changes to the code?

**Comment by davidmoraleees** [comment]
Exactly, the code is only prepared for right breast images. This was done to a lack of time. However, it can easily be programmed for left CC and MLO images. You should have a look into `find_farthest_point_from_chest_wall(...)`, create a new function that finds the nearest left point, and maybe you should revise `calculate_slope(...)` and `find_intersection(...)` to make sure you don't have any problems.

**Comment by davidmoraleees** [comment]
One question, why are you interested in this project? Are you working on something similar? Do you plan to use any part of this project in your work? If so, please make sure to clearly state that you are referencing this project in your paper and cite the authors (me and my colleague).

**Comment by emrekeles-arch** [comment]
For now, it's an individual project. I am trying to detect masses and calcifications. But if it turns into a professional work, I will definitely reference your work. Thank you again.

**Comment by davidmoraleees** [comment]
Okay! Let me know when you complete your project, as I am curious to see your results. Good luck!
